examples/flutter_notifications/lib/notification/bloc/notification_bloc.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter_notifications/notification/models/notification.dart';
+import 'package:notification_repository/notification_repository.dart';
+
+part 'notification_event.dart';
+part 'notification_state.dart';
+
+
+/// A Bloc that listens to [NotificationRepository]'s notificationStream, and add a [NotificationReceived] event
+/// in response to new notification received.
+class NotificationBloc extends Bloc<NotificationEvent, NotificationState> {
+  NotificationBloc({required NotificationRepository notificationRepository})
+      : _notificationRepository = notificationRepository,
+        super(NotificationInitial()) {
+    _notificationSubscription = _notificationRepository.notificationStream
+        .listen((NotificationEntity notificationEntity) {
+      add(NotificationReceived(notificationEntity));
+    });
+  }
+
+  final NotificationRepository _notificationRepository;
+  late StreamSubscription<NotificationEntity> _notificationSubscription;
+  @override
+  Stream<NotificationState> mapEventToState(
```

Comment:
As of v7.2.0, `mapEventToState` is deprecated. In order to merge this PR, you'll have to migrate to v7.2.0 or higher. You can check the migration guide [here](https://bloclibrary.dev/#/migration?id=v720).